indico/modules/events/registration/client/js/form/FormErrorList.jsx

```diff
@@ -0,0 +1,101 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useMemo} from 'react';
+import {useForm} from 'react-final-form';
+import {useSelector} from 'react-redux';
+
+import {Translate} from 'indico/react/i18n';
+
+import {getFlatFieldList} from './selectors';
+import './FormErrorList.module.scss';
+
+export default function FormErrorList() {
+  const formFieldList = useSelector(getFlatFieldList);
+  const fieldLabelLookup = useMemo(() => {
+    const lookup = new Map();
+    for (const formField of formFieldList) {
+      lookup.set(formField.htmlName, {
+        label: formField.title,
+        id: `input-${formField.id}`,
+      });
+    }
+    lookup.set('captcha', {
+      label: Translate.string('Captcha'),
+      id: 'input-captcha',
+    });
+    return lookup;
+  }, [formFieldList]);
+
+  const {errors} = useForm().getState();
+  const fieldsWithErrors = [];
+
+  for (const fieldName in errors) {
+    const field = fieldLabelLookup.get(fieldName);
+
+    // Check for misbehaving fields. This is for developers.
+    if (!field) {
+      console.warn(
+        `FormErrorList: Field ${fieldName} is not registered with the final form and will not be shown in the error list`
+      );
+      continue;
+    }
+
+    fieldsWithErrors.push({label: field.label, id: field.id, message: errors[fieldName]});
```

Comment:
Could be simplified to `{...field, message: errors[fieldName]}` since you use everything from `field` anyway.